src/api/autorun.ts

```diff
@@ -98,7 +98,7 @@ export function when(arg1: any, arg2: any, arg3?: any, arg4?: any) {
 		if (predicate.call(scope)) {
 			r.dispose();
 			const prevUntracked = untrackedStart();
-			effect.call(scope);
+			(effect as any).call(scope);
```

Comment:
These need to go. Had to do this get build working for now. TSC is not happy about current state of master when targeting ES6:


```
.build/mobx.ts(318,4): error TS2684: The 'this' context of type 'Lambda' is not assignable to method's 'this' of type 'Function'.
.build/mobx.ts(4127,10): error TS2684: The 'this' context of type 'Lambda' is not assignable to method's 'this' of type 'Function'.
  Property 'name' is optional in type 'Lambda' but required in type 'Function'.

```